libs/langchain/langchain/vectorstores/google/generativeai/base.py

```diff
@@ -0,0 +1,353 @@
+"""Google Generative AI Vector Store.
+
+The GenAI Semantic Retriever API is a managed end-to-end service that allows
+developers to create a corpus of documents to perform semantic search on
+related passages given a user query. For more information visit:
+https://developers.generativeai.google/guide
+"""
+
+import asyncio
+from functools import partial
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+    cast,
+)
+
+from langchain.schema.document import Document
+from langchain.schema.embeddings import Embeddings
+from langchain.schema.vectorstore import VectorStore
+
+VST = TypeVar("VST", bound="GoogleVectorStore")
+_import_err_msg = (
+    "`google.generativeai` package not found, "
+    "please run `pip install google-generativeai`"
+)
+
+
+class ServerSideEmbedding(Embeddings):
+    """Do nothing embedding model where the embedding is done by the server."""
+
+    def embed_documents(self, texts: List[str]) -> List[List[float]]:
+        return [[] for _ in texts]
+
+    def embed_query(self, text: str) -> List[float]:
+        return []
+
+
+class DoesNotExistsException(Exception):
+    def __init__(self, *, corpus_id: str, document_id: Optional[str] = None) -> None:
+        if document_id is None:
+            message = f"No such corpus {corpus_id}"
+        else:
+            message = f"No such document {document_id} under corpus {corpus_id}"
+        super().__init__(message)
+
+
+class GoogleVectorStore(VectorStore):
+    """Google GenerativeAI Vector Store.
```

Comment:
there's some inconsistent references in the docstrings: GenerativeAI Vector Store vs. Google Semantic Retriever.

WDYT if we stick to the same name across the comments: Maybe something like:
1. GenerativeAI Semantic Retriever (VectorStore)
2. GenerativeAI Vector Store (Semantic Retriever API)